prog.go

```diff
@@ -675,24 +680,37 @@ func (p *Program) testRun(opts *RunOptions) (uint32, time.Duration, error) {
 		Cpu:         opts.CPU,
 	}
 
+	if attr.Repeat == 0 {
+		attr.Repeat = 1
+	}
+
+retry:
 	for {
 		err := sys.ProgRun(&attr)
 		if err == nil {
-			break
+			break retry
 		}
 
 		if errors.Is(err, unix.EINTR) {
+			if attr.Repeat == 1 {
+				// Signal interruption is checked after each repetition on older
```

Comment:
What about kernels post-patch? The difference should be mentioned here. From my reading of the linked commit, signals are checked in `bpf_test_timer_continue()` as the `do..while` condition, so interleaved in between executions, never before starting execution. Doesn't that still qualify as 'after each repetition'?

Reply:
Rephrased.